irr/include/matrix4.h

```diff
@@ -1098,6 +1106,16 @@ inline vector3d<T> CMatrix4<T>::scaleThenInvRotVect(const vector3d<T> &v) const
 	};
 }
 
+template <class T>
+inline vector3d<T> CMatrix4<T>::transformVect(const vector3d<T> &v) const
+{
+	return {
+		v.X * M[0] + v.Y * M[4] + v.Z * M[8] + M[12],
+		v.X * M[1] + v.Y * M[5] + v.Z * M[9] + M[13],
+		v.X * M[2] + v.Y * M[6] + v.Z * M[10] + M[14],
+	};
```

Comment:
Could we perhaps avoid further code duplication by calling this new one in `void CMatrix4<T>::transformVect(vector3df &vect) const` below? The function is inline, which will likely result in identical code.

Reply:
sure. https://github.com/minetest/minetest/pull/15542/commits/97c845dfe7c89484d3dcf69a490065d07530c180